File: EpiFib/mobile/EpiFib/EpiFib/EmergencyResponderPacket.swift
//
//  EmergencyResponderPacket.swift
//  EpiFib
//
//  Created by Gregory Hewett on 3/19/17.
//  Copyright © 2017 The Reach Lab, LLC. All rights reserved.
//

import UIKit
import EVReflection

final class EmergencyResponderPacket: EVNetworkingObject
{
    var emergencyInstanceId: String?
    
    var emergencyType: EmergencyType?
    
    var ownerLocation: ExactLocation?
    
    var updatedTime: NSDate?
    
    var createdTime: NSDate?
    
    var ownerInfo: EmergencyUserInfo?
    
    convenience init(
        emergencyInstanceId: String,
        emergencyType: EmergencyType,
        ownerLocation: ExactLocation,
        updatedTime: NSDate,
        createdTime: NSDate,
        ownerInfo: EmergencyUserInfo)
    {
        self.init()
        self.emergencyInstanceId = emergencyInstanceId
        self.emergencyType = emergencyType
        self.ownerLocation = ownerLocation
        self.updatedTime = updatedTime
        self.createdTime = createdTime
        self.ownerInfo = ownerInfo
    }
}

File: EpiFib/www/src/Common/Models/EmergencyResponderPacket.cs
// <copyright file="EmergencyResponderPacket.cs" company="The Reach Lab, LLC">
//     The Reach Lab, LLC. All rights reserved.
// </copyright>
// <author>Gregory Hewett</author>

namespace Common.Models
{
    using System;

    public class EmergencyResponderPacket : IEmergencyResponderPacket
    {
        public string EmergencyInstanceId { get; set; }

        public EmergencyType EmergencyType { get; set; }

        public ExactLocation OwnerLocation { get; set; }

        public DateTime UpdatedTime { get; set; }

        public DateTime CreatedTime { get; set; }

        public EmergencyUserInfo OwnerInfo { get; set; }

        public static explicit operator EmergencyResponderPacket(EmergencyInstance v)
        {
            return new EmergencyResponderPacket
            {
                EmergencyInstanceId = v.EmergencyInstanceId,
                EmergencyType = v.EmergencyType,
                OwnerLocation = v.OwnerLocation,
                OwnerInfo = v.OwnerInfo
            };
        }
    }
}
